### PR TITLE
Revive and update printing processing tests

### PR DESCRIPTION
**Pull request by Ser-Zar**
#194
This PR brings back the abandoned tests from #153, brings them up to date and addresses all the feedback from the previous reviews.

List of changes:
Full Pytest Migration - Dropped the old `unittest` syntax and the hacky `sys.modules` Django mocks. We're using standard test settings now.
Better Test Isolation - Replaced the hardcoded `/tmp` paths with pytest's dynamic `tmp_path` fixture. This ensures each test gets its own unique directory
More Edge Cases - Added parameterized tests to cover weird page ranges (like overlapping or reversed numbers `3,2,1`), extreme page size ratios (squares, very long pages), and odd-page booklet impositions.
Cleanup - Removed accidentally committed `.coverage` files, added them to `.gitignore`, and made sure all files end with a proper trailing newline.

### CODE REVIEW

File: backend/.coverage
Comment (dominik-korsa) [inline backend/.coverage:1]: Coverage files should be excluded in `.gitignore`

File: backend/.coverage 2
Comment (dominik-korsa) [inline backend/.coverage 2:1]: Coverage files should be excluded in `.gitignore`

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ The changes below are formatted according to [keep a changelog].
 See also [Creating new releases] for instructions on how to create a new release.
 
 ## [Unreleased]
+### Changed
+- Migrated printing processing tests from `unittest` to `pytest`, parameterized test cases, and removed redundant `sys.modules` Django mocking [#153]
+### Added
+- Added new edge-case tests for page size ratio orientations, booklet odd-page imposition, and disabled fit-to-page scaling [#153]
```
Comment (dominik-korsa) [inline CHANGELOG.md]: Wrong PR number

File: backend/control/test_validators.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        "abc",
+        "0",
+        "3-1",
+        "1,,2",
+        "1-2-3",
+    ],
+)
+def test_validate_pages_to_print_rejects_invalid_formats(value):
+    with pytest.raises(ValidationError):
+        validate_pages_to_print(value)
```
Comment (dominik-korsa) [inline backend/control/test_validators.py:37]: Add tests for overlapping ranges and unusual orders. These tests are not comprehensive at all.

File: backend/printing/tests/printing/processing/test_final_pages.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+class TestFinalPageProcessor:
+
+    def test_init_perfect_square(self, mock_page_sizes):
+        fp = FinalPageProcessor("/tmp", 4, mock_page_sizes, PageOrientation.PORTRAIT, True)
+        assert fp.columns == 2
+        assert fp.rows == 2
+
+    def test_init_perfect_square_times_two(self, mock_page_sizes):
+        fp = FinalPageProcessor("/tmp", 2, mock_page_sizes, PageOrientation.PORTRAIT, True)
```
Comment (dominik-korsa) [inline backend/printing/tests/printing/processing/test_final_pages.py]: Dynamically create unique temporary directories for each test to keep them independent. Check if there is a standard way to do this in `pytest`.

File: backend/printing/tests/printing/processing/test_final_pages.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+        mock_reader.pages = [mock_page]
+        mock_dest_page = MagicMock()
+        mock_writer.add_blank_page.return_value = mock_dest_page
+
+        fp = FinalPageProcessor("/tmp", 1, mock_page_sizes, PageOrientation.PORTRAIT, fit_to_page=True)
+        res = fp.create_final_pages("in.pdf", "1")
+        assert "final_pages.pdf" in res
+
+    # NEW TEST 2: Ensures fit_to_page=False bypasses scaling entirely
```
Comment (dominik-korsa) [inline backend/printing/tests/printing/processing/test_final_pages.py]: What does *NEW TEST 2* mean here. Avoid using terms such as *new*, *old*, *latest* etc. in code, as they tend to get outdated quickly

**Review by dominik-korsa — CHANGES_REQUESTED** [review]

File: backend/printing/tests/printing/processing/test_final_pages.py (hunk 4 above)
Comment (Ser-Zar) [reply backend/printing/tests/printing/processing/test_final_pages.py]: I have already changed the name of the tests without unnecessary additions that might be outdated rapidly

File: CHANGELOG.md (hunk 1 above)
Comment (Ser-Zar) [reply CHANGELOG.md]: Already fixed

File: backend/.coverage
Comment (Ser-Zar) [reply backend/.coverage:1]: Done

File: backend/.coverage 2
Comment (Ser-Zar) [reply backend/.coverage 2:1]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (Ser-Zar) [reply CHANGELOG.md]: Already fixed

File: backend/control/test_validators.py (hunk 2 above)
Comment (Ser-Zar) [reply backend/control/test_validators.py:37]: I've included parameterized edge cases for overlapping ranges (like "1-5,3-7") and reversed/unusual orders (like "3,2,1" and "5-7,1-3")

File: backend/printing/tests/printing/processing/test_final_pages.py (hunk 3 above)
Comment (Ser-Zar) [reply backend/printing/tests/printing/processing/test_final_pages.py]: I've replaced the hardcoded "/tmp" strings with pytest's built-in tmp_path fixture. It automatically provisions and cleans up isolated directories for each test now.

File: backend/control/test_validators.py (hunk 2 above)
Comment (dominik-korsa) [reply backend/control/test_validators.py:37]: I would also test these in `test_pages_to_print_iter_range`

Comment (Ser-Zar) [reply backend/control/test_validators.py:37]: Ok added

File: backend/printing/tests/printing/processing/test_final_pages.py (hunk 3 above)
Comment (dominik-korsa) [reply backend/printing/tests/printing/processing/test_final_pages.py]: There are still a lot of cases where `/tmp`, `/tmp/sandbox` or similar dirs are used

Comment (Ser-Zar) [reply backend/printing/tests/printing/processing/test_final_pages.py]: Replaced the last static /tmp/sandbox with a dynamically generated tmp_path directory

**Review by dominik-korsa — APPROVED** [review]
